docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+### Goals
+
+- Implement immutable backup support for ETCD clusters.
+- Secure backup data against unintended or unauthorized modifications after creation.
+- Ensure backups are consistently available and intact for restoration purposes.
+
+### Non-Goals
+
+- Altering existing backup processes beyond what's necessary for immutability.
+- Implementing object-level immutability policies at this stage.
+- Supporting immutable backups on storage providers that do not offer immutability features (e.g., OpenStack Swift).
+
+## Proposal
+
+### Overview
+
+Introduce immutability in backup storage by leveraging cloud provider features that support a write-once-read-many (WORM) model. This will prevent data alterations after backup creation, enhancing data integrity and security. The implementation will focus on bucket-level immutability policies, as they are widely supported and easier to manage across different cloud providers.
+
+### Detailed Design
+
+#### Bucket Immutability Mechanism
+
+The Bucket Immutability feature configures an immutability policy for a cloud storage bucket, governing how long objects in the bucket must be retained in an immutable state. It also allows for locking the bucket's immutability policy, permanently preventing the policy from being reduced or removed.
+
+- **Supported by Major Providers:**
+  - **Google Cloud Storage (GCS):** [Bucket Lock](https://cloud.google.com/storage/docs/bucket-lock)
+  - **Amazon S3 (S3):** [Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html)
+  - **Azure Blob Storage (ABS):** [Immutable Blob Storage](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-policy-configure-container-scope?tabs=azure-portal)
+
+- **Not Supported:**
+  - **OpenStack Swift**
+
+**Implementation Details:**
+
+- Operators are responsible for creating or updating buckets with these immutability settings before ETCD Druid starts uploading snapshots.
+- Once the bucket is configured to be immutable, snapshots uploaded by `etcd-backup-restore` are also immutable and cannot be altered or deleted until the immutability period expires.
+- No additional configuration needs to be passed to etcd-druid.
+
+#### ETCD Backup Configuration
+
+Operators need to ensure that the ETCD backup configuration aligns with the immutability requirements. This includes setting appropriate immutability periods.
+
+
+#### Handling of Hibernated Clusters
+
+In scenarios where an ETCD cluster has been hibernated (scaled down to zero replicas) for a duration that exceeds the immutability period, backups may become mutable again (behaviour depends on cloud provider [refer](#comparison-of-storage-provider-properties-for-bucket-level-and-object-level-immutability)), compromising the intended immutability guarantees.
```

Comment:
We hibernate shoot clusters and not ETCD cluster. 
So I should we rather state that, If you are using ETCD-Druid with Gardener and are consuming Gardener feature of Shoot Hibernation then we need to explicitly handle hibernated clusters. 

We can also mention the behavior and expectation of standalone (without Gardener) usage of etcd-druid and how similar behavior can be achieved.

Reply:
We now state that an etcd cluster can now be "hibernated", and the following is meant by hibernation:

When an etcd cluster is intended to be scaled in and stopped from serving traffic, **the spec of the etcd can be changed** to have a field `hibernated: true`. 
etcd-druid ensures that the necessary backups have been taken, and scales-in the cluster. 
When the spec has `hibernated: true` removed, or set to `hibernated: false`, the next reconciliation will recover the etcd cluster from the backups.

The following are the implications:

- Consuming Gardener: When a shoot cluster is hibernated, then the corresponding etcd cluster is also hibernated by etcd-druid.
- Standalone: When an etcd cluster is to be hibernated, the spec of the etcd is to be changed by an operator.